### PR TITLE
Update GHA workflows to enforce FIMS installation

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
# What is the feature?

Update GHA workflows to ensure successful checks on the main branch.

# How have you implemented the solution?

- Modify `run-googletest.yml` and `get-gtest-codecov.yml` to enforce FIMS installation.
- add PAT to env variable `GITHUB_PAT` in both run-googletest.yml and get-gtest-codecov.yml

# Does the PR impact any other area of the project?

# How to test this change

All GHA checks have passed, see details [here](https://github.com/NOAA-FIMS/FIMS/actions?query=branch%3Afix-GHA)

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

File: .github/workflows/get-gtest-codecov.yml [hunk 1]
```diff
@@ -32,7 +34,7 @@ jobs:
       with:
         extra-packages: any::remotes
     - name: install local version of FIMS
-      run: remotes::install_local(upgrade = "always")
+      run: remotes::install_github("NOAA-FIMS/FIMS", upgrade = "always", force = TRUE)
```
Comment (k-doering-NOAA) [inline .github/workflows/get-gtest-codecov.yml:37]: I think this always install the main branch, so the code coverage won't be accurate for branches, which I don't think is what we want?

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA , thanks for working on this!

Hmm, it looks like these workflows now are only testing code on the main branch, though, which I don't think is what we want? Maybe we need to find a way to specify the branch in the install statement? (I also don't understand the differences between install_github and install_local, and why one works and one doesnt...)

I also wonder if getting rid of the install dependencies step would help; I don't think it is needed if using install_github()

I'm happy to look into options with you to get these working again!

**Comment by k-doering-NOAA** [comment]
what about using `install_local(upgrade= "always", force = TRUE)`?

**Comment by Bai-Li-NOAA** [comment]
@k-doering-NOAA I think `install_local(upgrade= "always", force = TRUE)` may solve the issue. I will give it a try and let you know!

File: .github/workflows/get-gtest-codecov.yml (hunk 1 above)
Comment (Bai-Li-NOAA) [reply .github/workflows/get-gtest-codecov.yml]: Thanks @k-doering-NOAA , the issue was resolved using `install_local(upgrade= "always", force = TRUE)`.

Comment (k-doering-NOAA) [reply .github/workflows/get-gtest-codecov.yml]: fantastic!!!

**Review by k-doering-NOAA — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**